### DISCUSSION TITLE
Compiling Peacock myself always breaks everything

### DISCUSSION THREAD

**Discussion by thecatontheceiling** [thepeacockproject/Peacock#474] [Q&A]
Even if I add only one single space to one file, whenever I compile with `yarn install`, `yarn build` and `yarn optimize`, it always ends up breaking save functionality in some way.

![image](https://github.com/thepeacockproject/Peacock/assets/75037904/474cb20e-1404-499f-a05b-b41f783ab40a)
It spams these errors, usually when I've been compromised in-game but it can also happen in other situations. 

I have attached the outputted chunk0.js file here:
https://gofile.io/d/zzcqzT
https://gofile.io/d/pPTjwY

**Comment by grappigegovert** [comment]
Does this error also show up when you run Peacock using `yarn run-dev` instead of building it?

**Comment by thecatontheceiling** [comment]
No, that seems to work fine. Something goes wrong when building.

**Comment by thecatontheceiling** [comment]
Nevermind, Peacock still seems to act weird when running run-dev, a whooooole lot of "failed to save sessions"

**Reply by AnthonyFuller** [reply]
If it says it's not in the server's memory, then that's fine.

**Reply by grappigegovert** [reply]
That's not weird, that's because the game is sending session ids to peacock which were created on a different server.
Maybe we should get rid of that warning

**Reply by RDIL** [reply]
It’s a dev only warning I believe

**Reply by thecatontheceiling** [reply]
No, it appears in release builds as well lol, anyway there's other weird things i've noticed except that

**Reply by thecatontheceiling** [reply]
Okay nevermind, restart the server and it's fixed (?), so `yarn run-dev` works properly. Build is still broken

**Reply by AnthonyFuller** [reply]
> It’s a dev only warning I believe

It does occur in release and dev.

**Comment by grappigegovert** [comment]
Can you run `node chunk0.js tools`, export debug info and upload that here, after racking up some of these errors in your build?

**Reply by thecatontheceiling** [reply]
> Can you run `node chunk0.js tools`, export debug info and upload that here, after racking up some of these errors in your build?

https://gofile.io/d/L23Imo

**Reply by grappigegovert** [reply]
Did you make different changes to the code here than you did in the OP? The chunk0 checksums don't seem to match

**Reply by thecatontheceiling** [reply]
> Did you make different changes to the code here than you did in the OP? The chunk0 checksums don't seem to match

Yeah, but both chunk0's cause these errors to appear in the console, I didn't think what I changed would matter

**Comment by grappigegovert** [comment]
> Even if I add only one single space to one file

Does this error also occur if you do not make any changes to the code before building?

**Reply by thecatontheceiling** [reply]
Nope

**Reply by grappigegovert** [reply]
Can you give me the output of a `git diff` after building a breaking build?

**Reply by thecatontheceiling** [reply]
Uh.... just realized these errors only appear when I'm in sapienza, wtf? I've been doing all my tests on Sapienza specifically but now I tried other maps and those seem to work no issue. No errors in console no matter what I change

**Reply by thecatontheceiling** [reply]
Alright I tried compromising myself in Miami, Isle of sgail, the final test, paris and santa fortuna, only compromising myself in Sapienza causes these errors to occur.

I found the only reference to a.push, which is in controller.ts:
https://github.com/thepeacockproject/Peacock/blob/master/components/controller.ts#L704

I am not touching this thing at all in any of my tests. Is this a bug with Peacock?

"a.push is not a function" suggests a is not a array/dictionary/map/string

**Comment by RDIL** [comment]
I've seen this issue in prod builds before so I'm gonna say it's a bug and open a tracking issue for it.
